File: domain/src/main/java/com/domain/remote/usecase/AbstractObserver.java
package com.domain.remote.usecase;

import io.reactivex.observers.DisposableObserver;

/**
 * @author volkanhotur
 */

public abstract class AbstractObserver<T> extends DisposableObserver<T> {

    @Override
    public void onNext(T t) { }

    @Override
    public void onComplete() { }

    @Override
    public void onError(Throwable exception) { }
}

File: domain/src/main/java/com/domain/remote/usecase/AbstractObserver.kt
package com.domain.remote.usecase

import io.reactivex.observers.DisposableObserver

/**
 * @author volkanhotur
 */
abstract class AbstractObserver<T> : DisposableObserver<T>() {

    override fun onNext(t: T) {}

    override fun onComplete() {}

    override fun onError(exception: Throwable) {}
}